github/github.go

```diff
@@ -880,6 +880,32 @@ func cloneRequest(r *http.Request) *http.Request {
 	return r2
 }
 
+// formatRateLimitResetDuration formats d to look like "[rate reset in 2s]" or
+// "[rate reset in 87m02s]" for the positive durations. And like "[rate limit was reset 87m02s ago]"
+// for the negative cases.
+func formatRateReset(d time.Duration) string {
+	isNegative := d < 0
```

Comment:
One thought hit me that it might make this simpler and easier to read if `isNegative` was checked only once (and then didn't actually need to be created at all...  something like:

```go
fmtStr := "[rate reset in %v]"
if d < 0 {
	d = -d
	fmtStr = "[rate limit was reset %v ago]"
}
secondsTotal := int(0.5 + d.Seconds())
minutes := secondsTotal / 60
seconds := secondsTotal - minutes*60

var timeString string
if minutes > 0 {
	timeString = fmt.Sprintf("%dm%02ds", minutes, seconds)
} else {
	timeString = fmt.Sprintf("%ds", seconds)
}

return fmt.Sprintf(fmtStr, timeString)
```